external-crates/move/crates/move-package/src/compilation/compiled_package.rs

```diff
@@ -756,3 +765,123 @@ pub(crate) fn make_source_and_deps_for_compiler(
     };
     Ok((source_package_paths, deps_package_paths))
 }
+
+/// partitions `deps` by whether we need to compile dependent packages with a
+/// prior toolchain (which we find by looking at Move.lock contents) or
+/// whether we can compile them with the current binary.
+fn partition_deps_by_toolchain(
+    deps: Vec<PackagePaths>,
+) -> Result<(Vec<PackagePaths>, Vec<PackagePaths>)> {
+    let mut deps_for_current_compiler = vec![];
+    let mut deps_for_prior_compiler = vec![];
+    for dep in deps {
+        let a_source_path = dep.paths[0].as_str();
+        let root = SourcePackageLayout::try_find_root(Path::new(a_source_path))?;
+        let lock_file = root.join("Move.lock");
+        if !lock_file.exists() {
+            // Q: Behavior to pick when package has no lock file. Right now we choose the current compiler (we could use the "legacy" one instead).
+            deps_for_current_compiler.push(dep);
+            continue;
+        }
```

Comment:
Implementation detail: which compiler version makes sense to pick when there is no lock file (which may be the case for existing packages). My intuition is we go with the latest / current one.

Reply:
I think that makes sense? Though I don't fully understand under what situations we don't have a lock file

Reply:
I think any time we don't have a lock file, we should generate one? And if that is the case, yes, we should use the latest.

Reply:
Yep, we generate lock files if they don't exist. 

> situations we don't have a lock file

People might not check it in on a repo